### PR TITLE
[PKO-13] Automatic Big Package Support - add BinpackNextFitChunker

### PR DESCRIPTION
**Pull request by erdii**
Signed-off-by: Josh Gwosdz <jgwosdz@redhat.com>

### Summary

This PR adds a chunker that put's at most 1 Megabyte (1 MiB) of objects into a single ObjectSlice and makes PKO behave as described in [1]:

> When using the 'Package' API, Package Operator will automatically split packages that reach a certain limit by using the 'ObjectSlice' API.

The chunker uses next fit binpacking aka:
- keep 1 bucket open
- put objects into the open bucket
- close the bucket and create a new bucket if the next object is too big

### Change Type

Breaking Change
New Feature

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [x] Relevant documentation has been updated.
- [x] Clarify: will this chunker be used by default?

### Additional Information

[1] https://package-operator.run/docs/concepts/big-packages/

### CODE REVIEW

**Comment by erdii** [comment]
The council has decided to:
- Put the `.` in a new line.
- Enable the new chunker by default.

File: internal/packages/internal/packagedeploy/chunking.go [hunk 1]
```diff
@@ -44,8 +64,12 @@ func determineChunkingStrategyForPackage(pkg adapters.GenericPackageAccessor) ob
 	switch chunkingStrategy(strategy) {
 	case chunkingStrategyEachObject:
 		return &EachObjectChunker{}
-	default:
+	case chunkingStrategyBinpackNextFit:
+		return &BinpackNextFitChunker{}
```
Comment (kostola) [inline internal/packages/internal/packagedeploy/chunking.go:68]: nit: isn't this redundant given that the `default` branch also returns `&BinpackNextFitChunker{}`?

**Review by eqrx — DISMISSED** [review]

File: internal/packages/internal/packagedeploy/chunking.go (hunk 1 above)
Comment (eqrx) [reply internal/packages/internal/packagedeploy/chunking.go:68]: I find it actually good for understanding, the case is explicitly considered and by coincidence does the same as "anything else"

Comment (erdii) [reply internal/packages/internal/packagedeploy/chunking.go:68]: I'll keep it :)

**Review by eqrx — APPROVED** [review]

**Pull request merged**